### PR TITLE
fix(args): make args conversion stateless

### PR DESCRIPTION
**Pull request by hgwood**
Hello.

I was starting to make a contribution to cross-env and found this [test](https://github.com/kentcdodds/cross-env/blob/master/src/command.test.js#L21). It looks like it is buggy. `isWindowsMock` is set to `true`, but the test expects `$test` not to be converted. And it passes! :open_mouth: 

I looked into it and discovered this interesting fact about `RegExp.prototype.exec`: [it's not stateless when used with the `g` flag](http://stackoverflow.com/questions/1520800/why-regexp-with-global-flag-in-javascript-give-wrong-results)! That's what is making the test pass.

So I added this test:

```
expect(commandConvert('$test')).toBe(commandConvert('$test'))
```

It does fail on `master`. To fix the problem, I have removed the `g` flag from the regexps in `command.js`. It is not useful anyway because `exec` is only called once.

Thanks for making cross-env, and for your consideration.

### CODE REVIEW

File: src/command.test.js [hunk 1]
```diff
@@ -13,12 +13,20 @@ test(`converts unix-style env variable usage for windows`, () => {
...
 })
+
 test(`converts windows-style env variable usage for linux`, () => {
   isWindowsMock.__mock.returnValue = false
   expect(commandConvert('%test%')).toBe('$test')
 })
 
 test(`leaves variable unchanged when using correct operating system`, () => {
-  isWindowsMock.__mock.returnValue = true
+  isWindowsMock.__mock.returnValue = false
```
Comment (kentcdodds) [inline src/command.test.js]: Thank you!

File: src/command.test.js [hunk 2]
```diff
@@ -13,12 +13,20 @@ test(`converts unix-style env variable usage for windows`, () => {
...
   expect(commandConvert('$test')).toBe('$test')
 })
+
+test(`is stateless`, () => {
+  // this test prevents falling into regexp traps like this:
+  // http://bit.ly/1zWb3tn
```
Comment (kentcdodds) [inline src/command.test.js]: Could you use the full URL instead? http://stackoverflow.com/a/1520853/971592

Comment (hgwood) [inline src/command.test.js]: Sure! Tried to use http://stackoverflow.com/questions/1520800/why-regexp-with-global-flag-in-javascript-give-wrong-results/1520853#1520853 but it was too long for the 80 chars limit.

**Comment by hgwood** [comment]
All done.

**Pull request merged**